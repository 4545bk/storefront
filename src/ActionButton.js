import React, { forwardRef } from 'react'
import { styled } from '@mui/material/styles'
import { Button, Typography } from '@mui/material'
import PropTypes from 'prop-types'

const PREFIX = 'RSFActionButton'

const defaultClasses = {
  label: `${PREFIX}-label`,
  caption: `${PREFIX}-caption`,
  button: `${PREFIX}-button`,
  value: `${PREFIX}-value`,
}

const StyledButton = styled(Button)(({ theme }) => ({
  /**
   * Styles passed through to the [`Button`](https://mui.com/api/button/#css) element's
   * `label` CSS rule.
   */
  [`& .${defaultClasses.label}`]: {
    justifyContent: 'space-between',
    alignItems: 'baseline',
    textTransform: 'none',
  },

  /**
   * Styles applied to the label container.
   */
  [`& .${defaultClasses.caption}`]: {
    textTransform: 'none',
    fontWeight: 'bold',
  },

  /**
   * Styles passed through to the [`Button`](https://mui.com/api/button/#css) element's
   * `contained` CSS rule.
   */
  [`& .${defaultClasses.button}`]: {
    boxShadow: 'none',
    backgroundColor: theme.palette.grey[200],
  },

  /**
   * Styles applied to the values container.
   */
  [`& .${defaultClasses.value}`]: {
    color: theme.palette.text.primary,
    whiteSpace: 'nowrap',
    textOverflow: 'ellipses',
    marginLeft: '10px',
  },
}))

export {}

/**
 * This button class displays a label and value.
 *
 * Example:
 *
 * ```js
 *  <ActionButton label="Sort" value="Lowest Price" onClick={openSortMenu} />
 * ```
 */
const ActionButton = forwardRef(({ label, value, children, classes: c = {}, ...props }, ref) => {
  const classes = { ...defaultClasses, ...c }
  let { caption, value: valueClasses, button, label: labelClasses, ...otherClasses } = classes
  return (
    <StyledButton
      ref={ref}
      variant="contained"
      color="primary"
      classes={{
        contained: classes.button,
        label: classes.label,
        ...otherClasses,
      }}
      {...props}
    >
      <Typography variant="button" className={classes.caption}>
        {label}
      </Typography>
      <Typography variant="caption" className={classes.value}>
        {value}
      </Typography>
    </StyledButton>
  )
})

ActionButton.propTypes = {
  /**
   * Override or extend the styles applied to the component. See [CSS API](#css) below for more details.
   */
  classes: PropTypes.object,

  /**
   * The label to display on the left side of the button.
   */
  label: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),

  /**
   * The value to display on the right side of the button.
   */
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
}

export default ActionButton
