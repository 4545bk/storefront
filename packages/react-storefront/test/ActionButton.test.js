/**
 * @license
 * Copyright Â© 2017-2018 Moov Corporation.  All rights reserved.
 */
import React from 'react'
import { mount } from 'enzyme'
import ActionButton from '../src/ActionButton'

describe('ActionButton', () => {

  it('renders with no props', () => {
    const component = (
      <ActionButton/>
    )

    expect(mount(component)).toMatchSnapshot()
  })

  it('renders label and value', () => {
    const component = (
      <ActionButton label="Label" value="Value"/>
    )

    expect(mount(component)).toMatchSnapshot()
  })
  
})