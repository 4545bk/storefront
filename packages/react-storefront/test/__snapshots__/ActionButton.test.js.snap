// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionButton renders label and value 1`] = `
<WithStyles(ActionButton) label="Label" value="Value">
  <ActionButton label="Label" value="Value" classes={{...}}>
    <WithStyles(Button) variant="contained" classes={{...}}>
      <Button variant="contained" classes={{...}} color="default" component="button" disabled={false} disableFocusRipple={false} fullWidth={false} mini={false} size="medium" type="button">
        <WithStyles(ButtonBase) className="MuiButton-root-5 MuiButton-contained-16 MuiButton-raised-19 ActionButton-raised-1" disabled={false} focusRipple={true} focusVisibleClassName="MuiButton-focusVisible-24" component="button" type="button">
          <ButtonBase className="MuiButton-root-5 MuiButton-contained-16 MuiButton-raised-19 ActionButton-raised-1" disabled={false} focusRipple={true} focusVisibleClassName="MuiButton-focusVisible-24" component="button" type="button" classes={{...}} centerRipple={false} disableRipple={false} disableTouchRipple={false} tabIndex="0">
            <button className="MuiButtonBase-root-31 MuiButton-root-5 MuiButton-contained-16 MuiButton-raised-19 ActionButton-raised-1" onBlur={[Function]} onFocus={[Function]} onKeyDown={[Function]} onKeyUp={[Function]} onMouseDown={[Function]} onMouseLeave={[Function]} onMouseUp={[Function]} onTouchEnd={[Function]} onTouchMove={[Function]} onTouchStart={[Function]} tabIndex="0" type="button" disabled={false}>
              <span className="MuiButton-label-6 ActionButton-label-2">
                <WithStyles(Typography) variant="button" className="ActionButton-caption-3">
                  <Typography variant="button" className="ActionButton-caption-3" theme={{...}} classes={{...}} align="inherit" color="default" gutterBottom={false} headlineMapping={{...}} noWrap={false} paragraph={false}>
                    <span className="MuiTypography-root-34 MuiTypography-button-45 ActionButton-caption-3">
                      Label
                    </span>
                  </Typography>
                </WithStyles(Typography)>
                <WithStyles(Typography) variant="caption" className="ActionButton-value-4">
                  <Typography variant="caption" className="ActionButton-value-4" theme={{...}} classes={{...}} align="inherit" color="default" gutterBottom={false} headlineMapping={{...}} noWrap={false} paragraph={false}>
                    <span className="MuiTypography-root-34 MuiTypography-caption-44 ActionButton-value-4">
                      Value
                    </span>
                  </Typography>
                </WithStyles(Typography)>
              </span>
              <NoSsr defer={false} fallback={{...}}>
                <WithStyles(TouchRipple) innerRef={[Function]} center={false}>
                  <TouchRipple center={false} classes={{...}}>
                    <TransitionGroup component="span" enter={true} exit={true} className="MuiTouchRipple-root-69" childFactory={[Function: childFactory]}>
                      <span className="MuiTouchRipple-root-69" />
                    </TransitionGroup>
                  </TouchRipple>
                </WithStyles(TouchRipple)>
              </NoSsr>
            </button>
          </ButtonBase>
        </WithStyles(ButtonBase)>
      </Button>
    </WithStyles(Button)>
  </ActionButton>
</WithStyles(ActionButton)>
`;

exports[`ActionButton renders with no props 1`] = `
<WithStyles(ActionButton)>
  <ActionButton classes={{...}}>
    <WithStyles(Button) variant="contained" classes={{...}}>
      <Button variant="contained" classes={{...}} color="default" component="button" disabled={false} disableFocusRipple={false} fullWidth={false} mini={false} size="medium" type="button">
        <WithStyles(ButtonBase) className="MuiButton-root-5 MuiButton-contained-16 MuiButton-raised-19 ActionButton-raised-1" disabled={false} focusRipple={true} focusVisibleClassName="MuiButton-focusVisible-24" component="button" type="button">
          <ButtonBase className="MuiButton-root-5 MuiButton-contained-16 MuiButton-raised-19 ActionButton-raised-1" disabled={false} focusRipple={true} focusVisibleClassName="MuiButton-focusVisible-24" component="button" type="button" classes={{...}} centerRipple={false} disableRipple={false} disableTouchRipple={false} tabIndex="0">
            <button className="MuiButtonBase-root-31 MuiButton-root-5 MuiButton-contained-16 MuiButton-raised-19 ActionButton-raised-1" onBlur={[Function]} onFocus={[Function]} onKeyDown={[Function]} onKeyUp={[Function]} onMouseDown={[Function]} onMouseLeave={[Function]} onMouseUp={[Function]} onTouchEnd={[Function]} onTouchMove={[Function]} onTouchStart={[Function]} tabIndex="0" type="button" disabled={false}>
              <span className="MuiButton-label-6 ActionButton-label-2">
                <WithStyles(Typography) variant="button" className="ActionButton-caption-3">
                  <Typography variant="button" className="ActionButton-caption-3" theme={{...}} classes={{...}} align="inherit" color="default" gutterBottom={false} headlineMapping={{...}} noWrap={false} paragraph={false}>
                    <span className="MuiTypography-root-34 MuiTypography-button-45 ActionButton-caption-3" />
                  </Typography>
                </WithStyles(Typography)>
                <WithStyles(Typography) variant="caption" className="ActionButton-value-4">
                  <Typography variant="caption" className="ActionButton-value-4" theme={{...}} classes={{...}} align="inherit" color="default" gutterBottom={false} headlineMapping={{...}} noWrap={false} paragraph={false}>
                    <span className="MuiTypography-root-34 MuiTypography-caption-44 ActionButton-value-4" />
                  </Typography>
                </WithStyles(Typography)>
              </span>
              <NoSsr defer={false} fallback={{...}}>
                <WithStyles(TouchRipple) innerRef={[Function]} center={false}>
                  <TouchRipple center={false} classes={{...}}>
                    <TransitionGroup component="span" enter={true} exit={true} className="MuiTouchRipple-root-69" childFactory={[Function: childFactory]}>
                      <span className="MuiTouchRipple-root-69" />
                    </TransitionGroup>
                  </TouchRipple>
                </WithStyles(TouchRipple)>
              </NoSsr>
            </button>
          </ButtonBase>
        </WithStyles(ButtonBase)>
      </Button>
    </WithStyles(Button)>
  </ActionButton>
</WithStyles(ActionButton)>
`;
